consensus/src/block_storage/block_store.rs

```diff
@@ -96,6 +102,40 @@ pub struct BlockStore {
     time_service: Arc<dyn TimeService>,
 }
 
+pub fn update_counters_and_prune_blocks(
+    block_tree: Arc<RwLock<BlockTree>>,
+    storage: Arc<dyn PersistentLivenessStorage>,
+    root: Arc<ExecutedBlock>,
+    blocks_to_commit: &[Arc<ExecutedBlock>],
+    block_to_commit: Arc<ExecutedBlock>,
```

Comment:
this is confusing, why do we need this? isn't it the same blocks as above but with dummy executed value?

Reply:
yeah this is the last block.